go/common/query_types.go

```diff
@@ -26,6 +36,48 @@ type PublicTxData struct {
 	Finality        FinalityType
 }
 
+type PublicBatchListing struct {
+	BatchHeader
+}
+
+type PublicBlockListing struct {
+	BlockHeader types.Header `json:"blockHeader"`
+	RollupHash  common.Hash  `json:"rollupHash"`
+}
+
+// MarshalJSON custom marshals the RollupHeader into a json
+func (p *PublicBlockListing) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&struct {
+		RollupHash string       `json:"rollupHash"`
```

Comment:
Maybe create a 
```
type blockListingJson struct {
	RollupHash string       `json:"rollupHash"`
	Header     types.Header `json:"blockHeader"`
}
```
that both Marshal and Unmarshal can use to replace the anonymous structs? I'd remove the `json:` annotations from `PublicBlockListing` too since I guess they don't work on there.

Reply:
yeah, had a bunch of problems with the marshiling unmarshalling stuff. Let me clean that up !

Reply:
Ended up deleting :) 👏